lib/tasks/migrate.js

```diff
@@ -56,6 +56,12 @@ module.exports = function runMigrations(context) {
                 message: 'Invalid database username or password',
                 environment: context.instance.system.environment
             });
+        } else if (error.stdout && error.stdout.match(/npm install sqlite3 \-\-save/)) {
+            // We check stdout because knex outputs to stdout on this particular error
+            error = new errors.SystemError(
```

Comment:
All errors can either take an object or a string - the string is a shorthand if you just need an error message.

ConfigErrors have additional properties needed to add more help information, which is why an object is passed, whereas SystemErrors are simple enough to just require a string message.